// Copyright 2023 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message LandmarksTransformationCalculatorOptions {
  extend CalculatorOptions {
    optional LandmarksTransformationCalculatorOptions ext = 421309928;
  }

  // Normalize translation so that center of mass is in (0, 0, 0).
  message NormalizeTranslation {}

  // Flip axis by multiplying coordinates along it by `-1`.
  message FlipAxis {
    optional bool flip_x = 1 [default = false];
    optional bool flip_y = 2 [default = false];
    optional bool flip_z = 3 [default = false];
  }

  message Transformation {
    oneof transformation {
      NormalizeTranslation normalize_translation = 1;
      FlipAxis flip_axis = 2;
    }
  }

  // Transformations applied in given order.
  repeated Transformation transformation = 1;
}
